framework/i18n/Formatter.php

```diff
@@ -787,6 +787,51 @@ public function asRelativeTime($value, $referenceTime = null)
         }
     }
 
+    /**
+     * Formats the value as a time interval.
+     * @param \DateInterval|string|integer $value the value to be formatted, in ISO8601 format or number of seconds.
+     * @return string the formatted result.
+     */
+    public function asDuration($value)
+    {
+        if ($value === null) {
+            return $this->nullDisplay;
+        }
+        if ($value instanceof \DateInterval) {
+            $negative = $value->invert;
+            $interval = $value;
+        } elseif (is_numeric($value)) {
+            $negative = $value < 0;
+            $interval = new \DateInterval('PT' . abs($value) . 'S');
+        } elseif (strpos($value, '-') !== false) {
```

Comment:
`2016-08-09T18:31:42` is also a ISO8601-formatted time. So we need to fix PHPDoc